enhancements/siteconfig-extra-manifest-version-independence.md

```diff
@@ -0,0 +1,417 @@
+
+  * we will deprecate `extraManifestPath` starting in this release. If defined it will continue to function exactly as it does today, but customers will be recommended to use the new extraManifestSearchPaths field.
+
+  * support multiple entries in the list in 4.14 – for example, recommended version specific extra-manifests in one directory and customer's custom extra manifests in another directory
+
+  * multiple entries accumulate. All files from entry1 and all files from entry2 will be applied during site generation.
+
+
+* How does templating work with the git based flows?
+
+  * SiteGen will look for `.tmpl` files in specific directories as it currently does inside the ztp-container 
+
+* How does this affect the existing extra manifest filtering feature?
+
+  * We keep filtering and support as it is.
+  * The `extraManifestSearchPaths` collects up all the content and then passes to the filtering subsystem as it is today.
+
+
+* Do we support override of named files using the `extraManifestSearchPaths`?
+
+  * Yes. There are multiple directories in the list. Each entry adds more content to the set. If there are overlapping names, we will overwrite the previous filename.
+
+* Do we allow override of same named files in the old `extraManifestPath` field? 
+
+  * No, we do not. The old behavior does not change when customer define old `extraManifestPath` path and not the new field `extraManifestSearchPaths`.
+
+* What if customer include both `extraManifestPath` and `extraManifestSearchPaths` in the siteconfig?
+
+  * The behavior of `extraManifestSearchPaths` will take precedence. We will neither generate any CR from `extraManifestPath` nor the builtin CRs from the ztp-container
+
+
+
+### How does the proposal help
+
+
+* **Controllability:** The proposal provides the flexibility for the customer to include/exclude any CRs from multiple directories from user Git repository. 
+
+  * customer can put extra-manifests extracted from ztp-container at `myPath` and their own custom CRs in `myPath2` as shown below as an example 
+
+
+    ```yaml
+      - cluster:
+        extraManifestSearchPaths:  <------ new field
+          - myPath/
+          - myPath2/
+        extraManifests:
+          filter:			
+      ```
+
+  * Any same named CR file in `myPath2` will override from `myPath`. Prior to this change, the customer couldn't put same named CR in the extraManifestPath. Now overriding is possible but they have to opt in for `extraManifestSearchPaths`. 
+
+
+
+* **Version Independence**: 
+  * What is version independence in this context?
+    
+    * The ability to exclude the built in content and the user can set their system up to only ever pull from git. This means they can have a set of 4.12, 4.13, 4.14, ... extra manifests in git and point to the correct set based on what they are deploying (one tool supporting multiple versions).
```

Comment:
Why wouldn't we then deprecate the builtins and require that all CRs be stored in git vs this hybrid model ?

Reply:
We plan to deprecate builtins from 4.15 release, so for this release customer can have hybrid behavior. But, in the doc, we will recommend to put all the CRs in the git point them using `extraManifestSearchPaths`, but if they don't and choose to do the old way, sitegen will still be able to generate CRs.